File: spring-java/src/main/java/com/jmpark/springjava/repository/CrudRepository.java
package com.jmpark.springjava.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface CrudRepository<T, ID> {
    Optional<T> create();
    Optional<T> update();
    void delete();
    Optional<T> one(ID id);
    List<T> all();
    Page<T> page(Pageable pageable);
}

File: spring-kotlin/src/main/kotlin/com/jmpark/springkotlin/repository/CrudRepository.kt
package com.jmpark.springkotlin.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface CrudRepository<Any, ID> {
    fun create(): Any?
    fun update(): Any?
    fun delete()
    fun one(id: ID): Any?
    fun all(): List<Any>
    fun page(pageable: Pageable): Page<Any>
}